### PR TITLE
ocp4: fix basic-auth check

### PR DESCRIPTION
**Pull request by JAORMX**
The check is now done through an API check.

### CODE REVIEW

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 8e0c9d07af5e93a9d2b8d2e870da637a90ceae4b

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 1f7c5b0d76cdaa4e5dfc67e076429ce9c8c8f5c4

**Review by jhrozek — APPROVED** [review]
/lgtm

**Comment by jhrozek** [comment]
Oops, sorry, I might have acked too soon, RHEL-6 and 7 are complaining:
```
Exception while handling file: /home/jenkins/workspace/scap-security-guide-pull-requests/label/rhel7/build/ocp4/rules/api_server_basic_auth.yml
Traceback (most recent call last):
  File "/home/jenkins/workspace/scap-security-guide-pull-requests/label/rhel7/build-scripts/build_templated_content.py", line 56, in <module>
    builder.build()
  File "/home/jenkins/workspace/scap-security-guide-pull-requests/label/rhel7/ssg/templates.py", line 550, in build
    self.build_all_rules()
  File "/home/jenkins/workspace/scap-security-guide-pull-requests/label/rhel7/ssg/templates.py", line 528, in build_all_rules
    rule = ssg.build_yaml.Rule.from_yaml(rule_path, self.env_yaml)
  File "/home/jenkins/workspace/scap-security-guide-pull-requests/label/rhel7/ssg/build_yaml.py", line 878, in from_yaml
    yaml_contents = open_and_macro_expand(yaml_file, env_yaml)
  File "/home/jenkins/workspace/scap-security-guide-pull-requests/label/rhel7/ssg/yaml.py", line 126, in open_and_macro_expand
    return open_and_expand(yaml_file, substitutions_dict)
  File "/home/jenkins/workspace/scap-security-guide-pull-requests/label/rhel7/ssg/yaml.py", line 109, in open_and_expand
    yaml_contents = _open_yaml(expanded_template, yaml_file, substitutions_dict)
  File "/home/jenkins/workspace/scap-security-guide-pull-requests/label/rhel7/ssg/yaml.py", line 75, in _open_yaml
    raise e
yaml.constructor.ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/unicode'
  in "<unicode string>", line 40, column 15
gmake[2]: *** [ocp4/templated-content-ocp4] Error 1
gmake[1]: *** [ocp4/CMakeFiles/generate-internal-templated-content-ocp4.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[  1%] Built target generate-internal-fedora-shorthand.xml
```

**Comment by jhrozek** [comment]
Do we care about ^ ? Can't we just let RHEL-7 ride into the sunset?

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now e06bad6605d01bddd62742845d9aa852640752eb

**Comment by JAORMX** [comment]
> Do we care about ^ ? Can't we just let RHEL-7 ride into the sunset?

I don't think we care much about those.

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 49ea5a6696a7541556455895c3c629b6d3af67c2

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 3d9e2d6a70f8941964008a6fdf8a3944e6f8a244

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now e06bad6605d01bddd62742845d9aa852640752eb

File: applications/openshift/api-server/api_server_basic_auth/rule.yml [hunk 1]
```diff
@@ -48,8 +48,26 @@ ocil: |-
...
+template:
+    name: yamlfile_value
+    vars:
+        filepath: '/api/v1/namespaces/openshift-kube-apiserver/configmaps/config'
+        yamlpath: ".data['config.yaml']"
+        value: "basic-auth"
+        type: "string"
+        operation: "pattern match"
+        entity_check: "none satisfy"
+        ocp_data: true
```
Comment (yuumasato) [inline applications/openshift/api-server/api_server_basic_auth/rule.yml:73]: Please, make sure the files have a newline at the end of the file.

Comment (yuumasato) [reply applications/openshift/api-server/api_server_basic_auth/rule.yml:73]: I think the issue is that `ocp_data` should have value `"true"`, not `true`

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 473c7671a0808ca4a655077e1c9ac86c8551e9ed

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 896385bcde69e4f8eacda60b51fa7593c207b831

**Review by jhrozek — APPROVED** [review]
great, the python2 build passes now, ACK!

**Pull request merged**